library/src/main/java/com/otaliastudios/zoom/ZoomImageView.kt

```diff
@@ -0,0 +1,104 @@
+package com.otaliastudios.zoom
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Matrix
+import android.graphics.RectF
+import android.graphics.drawable.Drawable
+import android.support.annotation.AttrRes
+import android.util.AttributeSet
+import android.view.Gravity
+import android.view.MotionEvent
+import android.widget.ImageView
+import com.otaliastudios.zoom.ZoomApi.ZoomType
+
+
+/**
+ * Uses [ZoomEngine] to allow zooming and pan events to the inner drawable.
+ *
+ *
+ * TODO: support padding (from inside ZoomEngine that gets the view)
+ */
+@SuppressLint("AppCompatCustomView")
+class ZoomImageView
+@JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, @AttrRes defStyleAttr: Int = 0, val engine: ZoomEngine = ZoomEngine(context))
```

Comment:
We don't want the full constructor to be public.

So, the primary constructor should be private. Then we can have 3 secondary constructors that pass the engine instance to the primary & you can use delegation.